File: DSA/Tree/Right_View.cpp
vector<int> getLeftView(TreeNode<int> *root) {
  //    Write your code here
    if (root == NULL)
        return {};
    vector<int>leftView;
    queue<TreeNode<int>*>q;
    q.push(root);
    while(!q.empty())
    {
        int size = q.size();
        for(int i = 1;i<=size;i++)
        {
            TreeNode<int>* node = q.front();
            q.pop();
            if(i == size)
                leftView.push_back(node->data);
            s
            if(node->left != NULL)
                q.push(node->left);
            if(node->right != NULL)
                q.push(node->right);
        }

    }
    return leftView;
}

File: Python/Tree/right_view.py
from collections import deque
def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root == None:
            return []
        queue = deque()
        queue.append(root)
        level = []
        while queue:
            size = len(queue)
            for i in range(1,size + 1):
                node =  queue.popleft()
                val  = node.val
                if i == size:
                    level.append(val)
                
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return level